### ISSUE TITLE
RN 0.64 (relevance unknown) and RN elements 3.3 do not work in Typescript

### ISSUE THREAD

**Issue by djMax** [react-native-elements/react-native-elements#2822]
**Explain what you did (Required)**

Updated from react-native-elements 3.2.0 to 3.3.0 (have been using RN 0.64.0-rc.3 for some weeks now, without issue)

**Expected behavior (Required)**

A working build

**Describe the bug (Required)**

Lots of typescript errors.

```
../../node_modules/react-native-elements/dist/avatar/Accessory.d.ts(9,70): error TS2344: Type '"source" | "style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 85 more ... | "placeholderStyle"' does not satisfy the constraint '"source" | "style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 85 more ... | "theme"'.
  Type '"accessibilityComponentType"' is not assignable to type '"source" | "style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 85 more ... | "theme"'.
../../node_modules/react-native-elements/dist/avatar/Avatar.d.ts(33,83): error TS2344: Type '"source" | "style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 85 more ... | "placeholderStyle"' does not satisfy the constraint '"source" | "style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 85 more ... | "theme"'.
  Type '"accessibilityComponentType"' is not assignable to type '"source" | "style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 85 more ... | "theme"'.
../../node_modules/react-native-elements/dist/icons/Icon.d.ts(30,65): error TS2344: Type '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "brand"' does not satisfy the constraint '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "theme"'.
  Type '"accessibilityComponentType"' is not assignable to type '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "theme"'.
../../node_modules/react-native-elements/dist/list/ListItem.d.ts(35,87): error TS2344: Type '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "brand"' does not satisfy the constraint '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "theme"'.
  Type '"accessibilityComponentType"' is not assignable to type '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "theme"'.
../../node_modules/react-native-elements/dist/list/ListItemChevron.d.ts(3,74): error TS2344: Type '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "brand"' does not satisfy the constraint '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "theme"'.
  Type '"accessibilityComponentType"' is not assignable to type '"style" | "hitSlop" | "onLayout" | "testID" | "nativeID" | "hasTVPreferredFocus" | "tvParallaxProperties" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 60 more ... | "theme"'.
../../node_modules/react-native-elements/dist/slider/Slider.d.ts(51,5): error TS2416: Property 'componentDidUpdate' in type 'Slider' is not assignable to the same property in base type 'Component<SliderProps, SliderState, any>'.
  Type '(prevProps: any, force: boolean) => void' is not assignable to type '(prevProps: Readonly<SliderProps>, prevState: Readonly<SliderState>, snapshot?: any) => void'.
    Types of parameters 'force' and 'prevState' are incompatible.
      Type 'Readonly<SliderState>' is not assignable to type 'boolean'.
```

**To Reproduce (Required)**

I don't have a simple repro project yet, will work on it if it's not obvious from the errors above.

**Your Environment (Required):**

| software              | version |
| --------------------- | ------- |
| react-native-elements |    3.3.0     |
| react-native          |    0.64.0-rc.3     |

**Comment by djMax** [comment]
This is likely related to #2802

**Comment by yim-jj** [comment]
same problem.

"react-native": "~0.63.4",
"react-native-elements": "^3.3.0",

import {Icon} from "react-native-elements";

<Icon name="close" onPress={handlePressClose} />


Type '{ name: string; onPress: () => void; }' is missing the following properties from type 'Pick<IconProps, "type" | "name" | "style" | "backgroundColor" | "borderRadius" | "testID" | "color" | "onLayout" | "accessible" | "accessibilityActions" | "accessibilityLabel" | ... 59 more ... | "brand">': accessibilityComponentType, accessibilityTraitsts(2739)

**Comment by meridPL** [comment]
I join the problem. I have that same problem with TS. I waiting for solve this problem

**Comment by flyingcircle** [comment]
Can someone confirm if this issue still exists in 3.3.2

**Comment by yim-jj** [comment]
> Can someone confirm if this issue still exists in 3.3.2

It has been resolved from v3.3.1
thanks!
